### DISCUSSION TITLE
Are AUC measures for learners that don't predict `lp` scores?

### DISCUSSION THREAD

**Discussion by bblodfon** [mlr-org/mlr3proba#356] [Q&A] [answered]
Hi!

Based on the code below, we see that the example `ranger` survival learner has no output prediction of type `lp`, while all ROC AUCs measures supported currently need that particular type to be calculated:
``` r
library(mlr3proba)
#> Loading required package: mlr3
library(mlr3extralearners)

task = tsk('rats')
task$select(cols = c('litter', 'rx'))

ranger_lrn = lrn('surv.ranger')
p = ranger_lrn$train(task)$predict(task)
# no `lp` prediction
p
#> <PredictionSurv> for 300 observations:
#>     row_ids time status     crank     distr
#>           1  101  FALSE  7.971548 <list[1]>
#>           2   49   TRUE 12.999444 <list[1]>
#>           3  104  FALSE 12.999444 <list[1]>
#> ---                                        
#>         298   92  FALSE  2.416444 <list[1]>
#>         299  104  FALSE  7.436310 <list[1]>
#>         300  102  FALSE  7.436310 <list[1]>

p$score(msr('surv.uno_auc'), task, train_set = 1:task$nrow)
#> surv.uno_auc 
#>          NaN

# all ROC AUCs measures require `lp` prediction type
for (measure in mlr_measures$keys(pattern = '^surv.*auc')) {
  print(mlr_measures$get(measure)$predict_type)
}
#> [1] "lp"
#> [1] "lp"
#> [1] "lp"
#> [1] "lp"
```

<sup>Created on 2022-06-07 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Could `crank` be somehow used in this case instead of `lp` (setting `lp` == `crank`? - maybe doesn't make sense to do that in general though). Maybe consider supporting a more general ROC AUC score (if it exists, a research question :) that requires a relative risk or distribution prediction?

**Comment by RaphaelS1** [comment, marked as answer]
I love this question because it has an easy answer! They don't exist.

If you look at the documentation in the `survAUC` package (or any of the papers the code comes from) you'll see they either: 1) expect a linear predictor from a Cox model; or more generally 2) expect a one-to-one relationship between the predicted risk and the survival distribution (which basically translates to requiring a linear predictor).

However time-dependent C-indices exist, and we have an open issue for this here https://github.com/mlr-org/mlr3proba/issues/267

**Comment by bblodfon** [comment]
For future reference: in the literature people have been using the $-S(t)$ at a particular time point $t$ as a risk prediction (instead of the linear predictor `lp`) to calculate AUC(t) scores. See [Blanche's 2013 paper](https://doi.org/10.1002/sim.5958): the formulas use a general marker `M` whose intepretation is "larger values => higher risk" with the assumption that the marker is independent of the censoring distribution. So $-S(t)$ can fit this description.

In this related discussion => https://github.com/bcjaeger/aorsf-bench/issues/6, a similar approach is taken for the calculation of AUC(t) and a simple mean across times gives a summary (time-independent) AUC score. A more general weighted form for an integrated AUC exists, see [Lambert's paper](https://doi.org/10.1177/0962280213515571) and MLSA book's [AUC chapter]( https://www.mlsabook.com/06-discrimination.html#sec-auc). Lastly, a direct application from Lambert's paper is the weighted-by-KM formula (7) which I have seen used in more recent ML studies.
